File: phase-1-arrays/LC3/code.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map<char , int> map; 
        int l = 0 , maxlen = 0 ; 

        for(int r = 0 ; r < s.size() ; r++)
        {
            if(map.count(s[r]) && map[s[r]] >= l)
            {
                l = map[s[r]] + 1 ; 
            }
            map[s[r]] = r; 
            maxlen = max(maxlen , r  - l  + 1);  
        }
        return maxlen; 
    }
};

File: phase-1-arrays/LC3/code.cs
using System; 

using System.Collections.Generic ; 

class Program
{
    static int LengthOfLongestSubstring(string s)
    {
        Dictionary<char , int> map = new Dictionary<char , int>(); 
        
        int l = 0 , max = 0 ; 
        for(int r = 0 ; r < s.Length ; r++)
        {
            if(map.ContainsKey(s[r]) && map[s[r]] >= l)
            {
                l = map[s[r]] + 1 ; 
            }
            
            map[s[r]] = r; 
            max = Math.Max(max , r - l + 1 ); 
        }
        return max ; 
    }

static void Main()
{
    Console.WriteLine(LengthOfLongestSubstring("abcabcbb"));
}
}